### PR TITLE
improv: freshen up dpp::snowflake and dpp::managed

### PR DESCRIPTION
**Pull request by Mishura4**
This PR modernizes dpp::snowflake and dpp::managed a little bit, and makes them constexpr-compatible
API is unchanged

## Code change checklist

- [x] I have ensured that all methods and functions are **fully documented** using doxygen style comments.
- [x] My code follows the [coding style guide](https://dpp.dev/coding-standards.html).
- [x] I tested that my change works before raising the PR.
- [x] I have ensured that I did not break any existing API calls.
- [x] I have not built my pull request using AI, a static analysis tool or similar without any human oversight.

### CODE REVIEW

File: include/dpp/snowflake.h [hunk 1]
```diff
@@ -22,8 +22,10 @@
 #pragma once
 #include <dpp/export.h>
 #include <dpp/json_fwd.h>
-#include <stdint.h>
-#include <functional>
+#include <dpp/exception.h>
+#include <cstdint>
+#include <charconv>
```
[the remark this answers is not in the document]
Comment (braindigitalis) [reply include/dpp/snowflake.h:27]: fixeroni, pepperoni, macaroni.... balloney?

[the remark this answers is not in the document]
Comment (Jaskowicz1) [reply include/dpp/snowflake.h:27]: rigatoni

[the remark this answers is not in the document]
Comment (braindigitalis) [reply include/dpp/snowflake.h:27]: Simpsons fat tony

9k=

File: include/dpp/snowflake.h [hunk 2]
```diff
@@ -52,141 +53,204 @@ class DPP_EXPORT snowflake final {
...
+
+	/**
+	 * @brief Move a snowflake object
+	 */
+	constexpr snowflake(snowflake &&rhs) noexcept = default;
+
+	/**
+	 * @brief Construct a snowflake from an integer value
+	 *
+	 * @throw dpp::logic_exception on assigning a negative value. the function is noexcept if the type given is unsigned
```
Comment (Jaskowicz1) [inline include/dpp/snowflake.h:77]: Im gonna be that guy, you need a capital on "the", should be `... negative value. The function...`

File: include/dpp/snowflake.h [hunk 3]
```diff
@@ -52,141 +53,204 @@ class DPP_EXPORT snowflake final {
...
+	 *
+	 * @param rhs The snowflake to move from
 	 */
-	operator uint64_t() const;
+	constexpr dpp::snowflake &operator=(dpp::snowflake&& rhs) noexcept = default;
+
+	/**
+	 * @brief Assign integer value to the snowflake
+	 *
+	 * @throw dpp::logic_exception on assigning a negative value. the function is noexcept if the type given is unsigned
```
Comment (Jaskowicz1) [inline include/dpp/snowflake.h:130]: ditto line 77

File: src/dpp/snowflake.cpp [hunk 4]
```diff
@@ -20,73 +20,28 @@
...
 namespace dpp {
 
-snowflake::snowflake(const uint64_t &value) : value(value) {}
-
-snowflake::snowflake(const std::string &string_value) {
-	try {
-		value = std::stoull(string_value);
-	}
-	catch (const std::exception &) {
+snowflake::snowflake(std::string_view string_value) noexcept {
```
Comment (Jaskowicz1) [inline src/dpp/snowflake.cpp:28]: All of this inside `namespace dpp {` should be indented as [defined by the code convention](https://dpp.dev/coding-standards.html#:~:text=all%20code%20blocks%20delimited%20within%20curly%20braces%20should%20be%20indented%20neatly%20and%20uniformly.).

**Review by Jaskowicz1 — CHANGES_REQUESTED** [review]

File: src/dpp/snowflake.cpp (hunk 4 above)
Comment (braindigitalis) [reply src/dpp/snowflake.cpp:28]: we don't indent the dpp namespace as it would just result in the entire file being indented 8 chars to the right...

Comment (braindigitalis) [reply src/dpp/snowflake.cpp:28]: (this should perhaps be added to the coding style)

File: include/dpp/snowflake.h (hunk 3 above)
Comment (Mishura4) [reply include/dpp/snowflake.h:130]: Will fix those when i get back to my pc

Comment (Jaskowicz1) [reply include/dpp/snowflake.h:130]: Sounds good :)

File: src/dpp/snowflake.cpp (hunk 4 above)
Comment (Jaskowicz1) [reply src/dpp/snowflake.cpp:28]: Then utility, and potentially more files, needs rewriting.

Comment (braindigitalis) [reply src/dpp/snowflake.cpp:28]: I think you misunderstood we don't and shouldn't indent these namespaces as they just push the entire file to the right

Comment (Jaskowicz1) [reply src/dpp/snowflake.cpp:28]: Agreed upon that the style guide needs to be changed to state that this case is correct.

**Review by braindigitalis — CHANGES_REQUESTED** [review]
namespace doesnt need to be indented any more

**Comment by Mishura4** [comment]
Updated

**Review by Jaskowicz1 — APPROVED** [review]

**Review by braindigitalis — APPROVED** [review]

**Pull request merged**